app/controllers/user_controller.rb

```diff
@@ -0,0 +1,25 @@
+class UserController < ApplicationController
+  def new
+  	@user = User.new
+  end
+
+  def create
+  	user = User.new(user_params)
+  	if user.save
+  		session[:user_id] = user.id
+  		redirect_to root_path
+  	else
+  		flash[:danger] = user.errors.messages.map do |key, value|
+  		redirect_to user_index_path
+  	end
+  end
```

Comment:
Hey Dan, I can't get the new user to be created in the database. If it's working for you I'll assume it's an issue on my end. Otherwise you could try this:
`def create
     @current_user.posts.create(post_params)
     redirect_to root_path
   end`